src/processors/modulation/Flanger.cpp

```diff
@@ -0,0 +1,297 @@
+#include "Flanger.h"
+#include "../BufferHelpers.h"
+#include "../ParameterHelpers.h"
+
+namespace
+{
+constexpr float rateLow = 0.5f;
+constexpr float rateHigh = 40.0f;
+constexpr float delayMs = 0.001f;
+
+const String delayTypeTag = "delay_type";
+} // namespace
+
+Flanger::Flanger (UndoManager* um) : BaseProcessor ("Flanger",
+                                                    createParameterLayout(),
+                                                    um,
+                                                    magic_enum::enum_count<InputPort>(),
+                                                    magic_enum::enum_count<OutputPort>())
+{
+    using namespace ParameterHelpers;
+    loadParameterPointer (rateParam, vts, "rate");
+    loadParameterPointer (delayAmountParam, vts, "delayAmount");
+    loadParameterPointer (delayOffsetParam, vts, "delayOffset");
+    loadParameterPointer (fbParam, vts, "feedback");
+    loadParameterPointer (mixParam, vts, "mix");
+    delayTypeParam = vts.getRawParameterValue (delayTypeTag);
+
+    addPopupMenuParameter (delayTypeTag);
+
+    uiOptions.backgroundColour = uiOptions.backgroundColour = Colour (106, 102, 190);
+    uiOptions.powerColour = Colours::yellow.brighter (0.1f);
+    uiOptions.info.description = "A flanger effect. Use the right-click menu to enable lo-fi mode.";
+    uiOptions.info.authors = StringArray { "Jatin Chowdhury and Kai Mikkelsen" };
+
+    routeExternalModulation ({ ModulationInput }, { ModulationOutput });
+    disableWhenInputConnected ({ "rate" }, ModulationInput);
+}
+
+ParamLayout Flanger::createParameterLayout()
+{
+    using namespace ParameterHelpers;
+    auto params = createBaseParams();
+
+    createPercentParameter (params, "rate", "Rate", 0.5f);
+    createTimeMsParameter (params, "delayAmount", "Amount", juce::NormalisableRange { 0.0f, 20.0f }, 2.0f);
+    createTimeMsParameter (params, "delayOffset", "Offset", juce::NormalisableRange { 0.0f, 20.0f }, 1.0f);
```

Comment:
I really like the default parameters that you've chosen! I wonder if we might want to skew the range so that the center value is something like 4-5 milliseconds?

Reply:
Good idea!